sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionContext.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   Contains information about a partition that an <see cref="IPartitionProcessor" /> will
+    ///   be processing events from.
+    /// </summary>
+    ///
+    public class PartitionContext
+    {
+        /// <summary>
+        ///   The path of the specific Event Hub that the context is associated with, relative
+        ///   to the Event Hubs namespace that contains it.
+        /// </summary>
+        ///
+        public string EventHubName { get; }
+
+        /// <summary>
+        ///   The name of the consumer group this context is associated with.
+        /// </summary>
+        ///
+        public string ConsumerGroup { get; }
+
+        /// <summary>
+        ///   The identifier of the associated <see cref="EventProcessor" /> instance.
+        /// </summary>
+        ///
+        public string InstanceId { get; }
+
+        /// <summary>
+        ///   The identifier of the Event Hub partition this context is associated with.
+        /// </summary>
+        ///
+        public string PartitionId { get; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="PartitionContext"/> class.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The path of the specific Event Hub this context is associated with, relative to the Event Hubs namespace that contains it.</param>
+        /// <param name="consumerGroup">The name of the consumer group this context is associated with.</param>
+        /// <param name="instanceId">The identifier of the associated <see cref="EventProcessor" /> instance.</param>
+        /// <param name="partitionId">The identifier of the Event Hub partition this context is associated with.</param>
+        ///
+        internal PartitionContext(string eventHubName,
+                                  string consumerGroup,
+                                  string instanceId,
+                                  string partitionId)
+        {
+            EventHubName = eventHubName;
+            ConsumerGroup = consumerGroup;
+            InstanceId = instanceId;
```

Comment:
What would `InstanceId` be used for?

Reply:
It's used by the `CheckpointManager` when creating/updating checkpoints. If we don't have it in the `PartitionContext`, there's no way for it to access the `ProcessorHost` id.

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/a596666e31cc0beb238adcf30201252af8508200/sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/CheckpointManager.cs#L64-L72

Reply:
Orthogonal to the review:  How'd you do that nifty think with the linking other code?

Reply:
You could pass the instanceId to the CheckpointManager constructor separately from the PartitionContext though right?

Since `PartitionContext` is going to be given to the customer directly, it seems like we should keep it simple. If there's a reason they would want the instance id though from within the `PartitionProcessor` then I could see keeping it in.

Reply:
@jsquire Just opened the file and copied a permalink to that code snippet.

Reply:
@chradek I was thinking about the user having access to the id, but given that the team has decided on exposing it in `EventProcessor`, passing it to the `CheckpointManager` constructor feels like a cleaner approach.